sklearn/cluster/optics_.py

```diff
@@ -562,10 +610,18 @@ def _extract_optics(ordering, reachability, maxima_ratio=.75,
         labels[index] = clustid
         is_core[index] = 1
         clustid += 1
+
+    # check if the last point in the ordering is a NOISE
+    last_point = ordering[-1]
+    if (core_distances_plot[-1] * 5 >= reachability_plot[-1]
+            and reachability_plot[-1] > reachability_plot[-2] * 5):
+        labels[last_point] = -1
+        is_core[last_point] = 0
```

Comment:
This is regarding having an outlier in the last position of `ordering_`. This checks for that the same way it checks if a split point is an outlier, since it would be a split point if it wasn't the last point in the `ordering_`.